File: day-06/python/part2.py
#!/usr/bin/env python3

import sys, string

data = [x.split() for x in open(sys.argv[1]).read().split("\n\n")]

total = 0

for q in data:
    u = set(string.ascii_lowercase)
    for c in q:
        u = u.intersection(set(c))
    
    total += len(u)
    
print(total)

File: day-06/nim/part2.nim
from os import paramStr
import strutils, sets

let data = paramStr(1).string.readfile.split("\n\n")

var total = 0

for q in data:
    var u: HashSet[char] = toHashSet(['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'])
    for c in q.split():
        u = u * toHashSet(c)

    total += len(u)

echo total
